### PR TITLE
doc & ci : modify README, upgrade docs and pr-commentor gh action for v3.11.0

### PR DESCRIPTION
**Pull request by Rakshith-R**
# Describe what this PR does #

- ci: update pr-commentor rules matrix
- doc: add steps for upgrading from 3.10 to 3.11
- doc: update README for v3.11.0 release

**Checklist:**

* [x] **Commit Message Formatting**: Commit titles and messages follow
  guidelines in the [developer
  guide](https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#commit-messages).
* [x] Reviewed the developer guide on [Submitting a Pull
  Request](https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#development-workflow)
* [x] [Pending release
  notes](https://github.com/ceph/ceph-csi/blob/devel/PendingReleaseNotes.md)
  updated with breaking and/or notable changes for the next major release.
* [x] Documentation has been updated, if necessary.
* [x] Unit tests have been added, if necessary.
* [x] Integration tests have been added, if necessary.

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

* `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)

</details>

### CODE REVIEW

**Review by Madhu-1 — APPROVED** [review]

File: .github/workflows/pull-request-commentor.yaml [hunk 1]
```diff
@@ -29,10 +29,6 @@ jobs:
           - k8s: "1.25"
             branch: "release-v3.10"
 
-          # Ceph-CSI <= 3.9 was released before Kubernetes 1.28
-          - k8s: "1.28"
-            branch: "release-v3.9"
-
           # Ceph-CSI <= 3.10 was released before Kubernetes 1.29
           - k8s: "1.29"
             branch: "release-v3.9"
```
Comment (Madhu-1) [inline .github/workflows/pull-request-commentor.yaml]: This can be removed as well.

File: .github/workflows/pull-request-commentor.yaml [hunk 2]
```diff
@@ -18,7 +18,7 @@ jobs:
 
     strategy:
       matrix:
-        branch: [release-v3.9, release-v3.10, devel]
+        branch: [release-v3.10, release-v3.11, devel]
         k8s: ["1.25", "1.26", "1.27", "1.28", "1.29"]
```
Comment (Madhu-1) [inline .github/workflows/pull-request-commentor.yaml]: Remove 1.25 Ref from here

**Review by Madhu-1 — CHANGES_REQUESTED** [review]
https://github.com/ceph/ceph-csi/blob/e92e4bf365b3894d104f2337919100688dee8195/.mergify.yml

here seems to be we are still referring 1.25 which can be removed

File: .mergify.yml [hunk 3]
```diff
@@ -343,15 +349,18 @@ pull_request_rules:
...
+              - "status-success=ci/centos/k8s-e2e-external-storage/1.28"
+              - "status-success=ci/centos/k8s-e2e-external-storage/1.29"
               - "status-success=ci/centos/mini-e2e-helm/k8s-1.26"
               - "status-success=ci/centos/mini-e2e-helm/k8s-1.27"
-              - "status-success=ci/centos/mini-e2e/k8s-1.25"
+              - "status-success=ci/centos/mini-e2e-helm/k8s-1.28"
+              - "status-success=ci/centos/mini-e2e-helm/k8s-1.29"
               - "status-success=ci/centos/mini-e2e/k8s-1.26"
               - "status-success=ci/centos/mini-e2e/k8s-1.27"
+              - "status-success=ci/centos/mini-e2e/k8s-1.28"
```
Comment (Madhu-1) [inline .mergify.yml:362]: https://github.com/ceph/ceph-csi/blob/45178c2a905e184f439fbc427609774d27a2c227/.mergify.yml#L151-L205

is not checking 1.26 for devel branch is that expected?

Comment (Rakshith-R) [reply .mergify.yml:362]: That's how the config is currently, I am not able to find the commit that removed 1.26 from that list.
do you want me to add that in this pr ?

Comment (Madhu-1) [reply .mergify.yml:362]: will revisit the mergify later, not required for now in that case

**Review by Madhu-1 — APPROVED** [review]

**Review by nixpanic — APPROVED** [review]

**Comment by nixpanic** [comment]
@mergifyio queue

**Pull request merged**